pkg/template/servicebroker/servicebroker.go

```diff
@@ -1,62 +1,99 @@
 package servicebroker
 
 import (
-	"k8s.io/apiserver/pkg/authentication/user"
+	"errors"
+	"fmt"
+	"time"
+
+	"github.com/golang/glog"
+
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
 	restclient "k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/cache"
 	kclientset "k8s.io/kubernetes/pkg/client/clientset_generated/internalclientset"
-	"k8s.io/kubernetes/pkg/client/clientset_generated/internalclientset/typed/core/internalversion"
 
-	authclient "github.com/openshift/origin/pkg/auth/client"
 	"github.com/openshift/origin/pkg/client"
-	"github.com/openshift/origin/pkg/client/cache"
-	"github.com/openshift/origin/pkg/controller/shared"
+	"github.com/openshift/origin/pkg/cmd/server/bootstrappolicy"
+	"github.com/openshift/origin/pkg/serviceaccounts"
+	templateapi "github.com/openshift/origin/pkg/template/api"
+	templateinformer "github.com/openshift/origin/pkg/template/generated/informers/internalversion/template/internalversion"
 	templateclientset "github.com/openshift/origin/pkg/template/generated/internalclientset"
 	internalversiontemplate "github.com/openshift/origin/pkg/template/generated/internalclientset/typed/template/internalversion"
+	templatelister "github.com/openshift/origin/pkg/template/generated/listers/template/internalversion"
 )
 
 type Broker struct {
-	secretsGetter      internalversion.SecretsGetter
-	localSAR           client.LocalSubjectAccessReviewsNamespacer
+	oc                 *client.Client
+	kc                 kclientset.Interface
 	templateclient     internalversiontemplate.TemplateInterface
 	restconfig         restclient.Config
-	lister             cache.StoreToTemplateLister
+	informer           cache.SharedIndexInformer
+	lister             templatelister.TemplateLister
 	templateNamespaces map[string]struct{}
+	ready              chan struct{}
 }
 
-func NewBroker(restconfig restclient.Config, localSAR client.LocalSubjectAccessReviewsNamespacer, secretsGetter internalversion.SecretsGetter, informers shared.InformerFactory, namespaces []string) *Broker {
+func NewBroker(privrestconfig restclient.Config, privkc kclientset.Interface, infraNamespace string, informer templateinformer.TemplateInformer, namespaces []string) *Broker {
+	informer.Informer().AddIndexers(cache.Indexers{
```

Comment:
Do this right where you create the informers.  We really shouldn't ever create one without this and later we can update the generator to allow it.